scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,546 @@
+    val artifactVersion = StringBuilder()
+    var lastIndex = artifactName.length - 1
+    while (lastIndex >= 0 && artifactName[lastIndex] != ':') {
+      artifactVersion.append(artifactName[lastIndex])
+      lastIndex--
+    }
+    artifactVersion.reverse()
+    val licenseNamesFromPom = mutableListOf<String>()
+    val licenseLinksFromPom = mutableListOf<String>()
+    val licenseList = arrayListOf<License>()
+    val pomFile = URL(pomFileUrl).openStream().bufferedReader().readText()
+    val pomText = pomFile
+    var cursor = -1
+    if (pomText.length > 11) {
+      for (index in 0..(pomText.length - 11)) {
+        if (pomText.substring(index, index + 10) == LICENSES_TAG) {
+          cursor = index + 9
+          break
+        }
+      }
+      if (cursor != -1) {
+        var cursor2 = cursor
+        while (cursor2 < (pomText.length - 12)) {
+          if (pomText.substring(cursor2, cursor2 + 9) == LICENSE_TAG) {
+            cursor2 += 9
+            while (cursor2 < pomText.length - 6 &&
+              pomText.substring(
+                cursor2,
+                cursor2 + 6
+              ) != NAME_TAG
+            ) {
+              ++cursor2
+            }
+            cursor2 += 6
+            val url = StringBuilder()
+            val urlName = StringBuilder()
+            while (pomText[cursor2] != '<') {
+              urlName.append(pomText[cursor2])
+              ++cursor2
+            }
+            while (cursor2 < pomText.length - 4 &&
+              pomText.substring(
+                cursor2,
+                cursor2 + 5
+              ) != URL_TAG
+            ) {
+              ++cursor2
+            }
+            cursor2 += 5
+            while (pomText[cursor2] != '<') {
+              url.append(pomText[cursor2])
+              ++cursor2
+            }
+            licenseNamesFromPom.add(urlName.toString())
+            licenseLinksFromPom.add(url.toString())
+            licenseList.add(
+              License
+                .newBuilder()
+                .setLicenseName(urlName.toString())
+                .setPrimaryLink(url.toString())
+                .setPrimaryLinkType(PrimaryLinkType.PRIMARY_LINK_TYPE_UNSPECIFIED)
+                .build()
+            )
+          } else if (pomText.substring(cursor2, cursor2 + 12) == LICENSES_CLOSE_TAG) {
+            break
+          }
+          ++cursor2
+        }
+      }
+    }
+    val mavenDependency = MavenDependency
+      .newBuilder()
+      .setArtifactName(it.coord)
+      .setArtifactVersion(artifactVersion.toString())
+      .addAllLicense(licenseList)
+      .setOriginOfLicense(OriginOfLicenses.UNKNOWN)
+
+    mavenDependencyList.add(mavenDependency.build())
+  }
+  return MavenDependencyList.newBuilder().addAllMavenDependency(mavenDependencyList).build()
+}
+
+fun writeTextProto(
+  pathToTextProto: String,
+  mavenDependencyList: MavenDependencyList
+) {
+  val file = File(pathToTextProto)
+  val list = mavenDependencyList.toString()
```

Comment:
We should be using TextFormat, not toString(). See the sample PR I shared with you that does this for our script to download assets for the alpha build.

Reply:
Looks like we don't need any.

Reply:
Sorry, I'm not following. Printing out a proto file the way you are isn't correct. We need to use TextFormat in order to ensure that the textproto file is compatible with the protoc rule we use to convert it to pb.

Reply:
Done.

Reply:
Verified.